### PR TITLE
DelaunayTriangulation example

### PR DESCRIPTION
**Pull request by DanielVandH**
As discussed @Datseris, this PR adds an example using DelaunayTriangulation.jl to simulate a cell model. The model has changed from what it used to be in DelaunayTriangulation.jl's, and now simulates interactions between three cell species and also includes cell death. This was my first time actually using the package so maybe some things are not done optimally.

Here's what the final video looks like.

https://github.com/user-attachments/assets/24f1b6fb-2e4d-46f9-8527-ed4934528379

### CODE REVIEW

File: docs/Project.toml
Comment (Tortar) [inline docs/Project.toml:16]:
```suggestion
LightSumTypes = "f56206fc-af4c-5561-a72a-43fe2ca5a923"
```

This package is actually deprecated in favour of `LightSumTypes.jl`

**Comment by Tortar** [comment]
Very cool model anyway!! My only concern is about plotting, is there a way to reuse more of the Agents.jl infrastructure for it?

**Comment by Datseris** [comment]
I'd like to review this before merging. Bit busy at the moment but will find time for sure.

**Comment by DanielVandH** [comment]
Maybe? I could have made the animation simpler and just not show the data graphs so that `abmvideo` works. But it seemed like a nice way to also demonstrate how that could be done.

If there's a way to change the recipe used in `abmplot` (e.g. something other than `scatter` or `poly`), then it will also be a lot simpler since I could use `voronoiplot`.

File: examples/delaunay.jl
Comment (Tortar) [inline examples/delaunay.jl:329]:
```suggestion
# Next, we need to know how to select a random triangle from a triangulation. This is done by using a weighted 
# sample according to the triangle areas.
using StreamSampling
function random_triangle(tri::Triangulation)
    triangles = DT.each_solid_triangle(tri)
    area(T) = DT.triangle_area(get_point(tri, triangle_vertices(T)...)...)
    T = itsample(triangles, area)
    return T
end
```

Comment (Tortar) [reply examples/delaunay.jl:329]: This package is already used in Agents, and it should reduce allocations, actually in this case maybe not, because I ran the example and I saw that `random_triangle` is used only a few times, but anyway this could be useful

Comment (Tortar) [reply examples/delaunay.jl]: yes, that is the reason, indeed I see some notable time improvements with `@time step!(initialize_cell_model(radius=0.1, ninit=3), 2000);`

**Review by Tortar — APPROVED** [review]

**Comment by Tortar** [comment]
To me, this seems mostly fine, just wait for the review of @Datseris to be merged :-)

**Comment by Tortar** [comment]
@Datseris do you have the time ro review this? Otherwise we could merge it and push updates in the future

**Comment by Datseris** [comment]
I'm sorry, I am in a difficult time in life now with no time to spare. I wanted to review because I am convinced we can simplify ploitting, but I don't have the time to do it now. GO ahead and merge and in the future maybe I can make a PR myself!

**Comment by Tortar** [comment]
no problem, I'm also running a bit out of time these days, after graduation I have a bit more frenetic routine :D 

If I find the time I will try to update the plotting part, but I don't want to put this onhold because I think it's a great contribution

**Pull request merged**
